### DISCUSSION TITLE
Proposal: Increase TApp consume action

### DISCUSSION THREAD

**Discussion by geopor** [tearust/teaproject#68] [Ideas]
### Proposal to change the consume amount for the daily YouTube views winner

- **Currently**: We consume into the bonding curve token of the YouTube video with the highest daily % gain by issuing a **10,000T** consume event.

- **Proposal**:  Three winners will be chosen every day. Their consume events are proposed to be:
  -  1st prize 12,000T
  -  2nd prize 6,000T
  -  3rd prize 3,000T

### Reason for proposed change
Participants in the mining contest can see their TApp asset values go up by actively promoting the associated videos. If someone spends effort to increase the number of views of a video but it doesn't reach the top spot for the day, that could leave them discouraged. We want to encourage curation and understand that our focus on % increase makes it hard for videos to compete for the single top spot day after day.

**Comment by kevingzhang** [comment]
Makes sense

**Reply by nick-yin** [reply]
同意提议

**Reply by Maotweety** [reply]
同意

**Comment by leeyemi** [comment]
Need to review the bonding curve.If I invest in crypto coins,.I don't mind if the value goes down but not both value and my coin quantity go down.
 I will  straight away pull out my investments.  And this  is what currently  happening.  Imagine you bought  10  btc then value goes down and  your btc  coins reduced  to  2btc.

**Comment by kevingzhang** [comment]
Your coin quantity goes down is because of the expense. Someone needs to pay for the hosting fee, and you also gain a hosting fee if your CML hosting these TApps. 

To avoid your token quantity goes down, you can either invest in a TApp that costs less to host, or invest in staking. 

In the real-life, a TApp costs less if it consumes less computing resources. This requires the developers have higher programming skills. Of course, this kind of TApp is an ideal investment target and usually values higher. On the other side, if a TApp costs a lot to host but earns less, this is a bad tApp, usually values less and likely bankruptcy. This is the market value of TApps.

**Comment by Maotweety** [comment]


**Reply by kevingzhang** [reply]
Yesterday, we have totally 6 tapps. 2 of them were dead overnight.

**Comment by Maotweety** [comment]


**Reply by kevingzhang** [reply]
English please.

**Reply by kevingzhang** [reply]
Please repost this message in English. thanks

**Comment by geopor** [comment]
Based on positive feedback we'll be implementing this feature immediately.

@leeyemi we understand the concern regarding the fluctuating number of tokens in wallets. Having three winners instead of only one  will help in that we are giving more **consume** events across more TApps which will boost the supply for token holders. A high initial token allotment can also help cushion the **expense** shocks so that is something we can adjust as we add more TApps. But as Kevin mentioned, the daily subtractions comes from token holders having to pay the hosting cost for TApps with high **hosting performance requirements** and if these TApps aren't getting any new token buyers or winning **consume** events then they will ultimately go to zero as the bonding curve gets depleted.

If you're curious about why some tokens contract faster than others in your wallet, please watch [the following video](https://www.youtube.com/watch?v=NSTx-k3Md9A).

**Comment by kevingzhang** [comment]
I would suggest changing the tipping rules.
- 5000T to the top growth rate
- 15000T distributed to all TApps based on their views growth weight. The higher the views growth in the past 24 hours, the higher the weight in the distribution. 

This can keep the investors stay in existing TApps instead of fleeing away after the tipping is completed every day.

**Comment by geopor** [comment]
We continue to test out different **consume** action amounts in an effort to boost participation in the TApp token economy. We want to make it profitable for miners to host TApps but know that their revenue comes from the **expense** action which is taken from token holders.

We're currently distributing 50,000T at various times throughout the day (currently 2-3 times) and allocating to each TApp's token bonding curve proportional to the number of views. If the TApp **TDEV** gets 20 views since the last measure point and all the TApps added together have gotten a net increase of 100 views, then TDEV's token would enjoy 20/100 share of the 50,000T rewards.

Going by the raw number increase makes it easier for videos with larger views to win a larger share of the distribution. A video with 1300 views could gain an extra 30 views during a period easily, while that would be much more difficult for a video with less than 100 views. 

**So we propose to adjust the distribution of the consume action reward as follows:**

- 50% of award is based on the % increase in views proportional to all TApps' % increase.
- 50% of award is based on the increase in number of views proportional to all the TApps' net increase in views added together.

The consume amount will stay at 50,000T per tipping round, with a variable number of rounds (currently 2-3) per day.

**Reply by kevingzhang** [reply]
Makes sense. let's try it for one day or two, see how it goes
